### PR TITLE
Allows OLM subcommands work for Helm/Ansible new layout

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**

- fix OLM subcommands to be able to get the operator name via the project name in the project file or the current directory.  
- add e2e tests to ensure that the bundle subcommands are working for projects scaffolds with Helm plugin and the new layout. 

**Motivation**

SDK is in a process to be integrated with KB which means that its project layouts will be aligned. More info : [Integrating Kubebuilder and Operator SDK](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/integrating-kubebuilder-and-osdk.md). 

#3327

### CODE REVIEW

**Comment by estroz** [comment]
Can you explain what is wrong with the current OLM-related commands other than path defaults? There might be different default paths for these project types but no need to change anything else within that scope (aside from the stuff relevant to #3295).

/hold

**Comment by camilamacedo86** [comment]
/hold until merge #3295 and  #3330 and then, we are able to rebase it with the master. 
The hold is just merged but the PR can be reviewed.

**Comment by camilamacedo86** [comment]
Hi @estroz, 

> Can you explain what is wrong with the current OLM-related commands other than path defaults? There might be different default paths for these project types but no need to change anything else within that scope (aside from the stuff relevant to #3295).

All is described in the first comment. See that: 
- The changes of this PR are just: https://github.com/operator-framework/operator-sdk/pull/3341/commits/d210134a49a30eb8ede4258f3de915c1c637a2e6 whcih as you described is just for we are able to get the OperatorName 
- We are making the commands use as OperatorName the current dir when the config.Repo do not exist

```go
// GetOperatorName will looking for the value in the config.repo and return the project dir if it do not exist
// Note that Ansible/Helm operators in the new layout do not have the repo information.
func GetOperatorName(cfg *config.Config) (string, error) {
	if len(strings.TrimSpace(cfg.Repo)) < 1 {
		// Check if the project name is a valid namespace according to k8s
		dir, err := os.Getwd()
		if err != nil {
			return "", fmt.Errorf("error to get the current path: %v", err)
		}
		return filepath.Base(dir), nil
	}
	return filepath.Base(cfg.Repo), nil
}
```

- We are adding the e2e tests for Helm scaffolded with the new layout to ensure that all OLM commands are working

**IMPORTANT**

This PR has as a blocker; 

 - PR which adds the helm plugin and the e2e tests that we are complementing here. See: #3295
 - PR that allows have the run command to exec operator-sdk run packagemanifests done in the e2e tests. See: #3330

**Comment by estroz** [comment]
While the concept of a repo (path) is unique to Go amongst all operator types the SDK supports, it can still be used by other operator types by setting it to the operator's name. This has the benefit of:
1. Allowing plugin/SDK code to assume all top-level mandatory fields for a particular PROJECT file are filled, like kubebuilder's Go plugin does.
1. Obviates the need to add `--operator-name` to commands, since the operator name will always be `path.Base(cfg.Repo)`, independent of operator type.

I'll re-review this once blocking PR's are merged.

**Comment by camilamacedo86** [comment]
Hi @estroz, 

So, your suggestion would be to add the repo in the PROJECT file for Ansible/Helm types as well. It was my first thought and solution implemented and I agree with the standard benefits raised by you. However, the @joelanford argumentation against it was that this information is not relevant to these types. 

@joelanford wdyt? Could we agree in keep the operator name in the repo project file instead of getting these value each time that it is required? See that it can be requested in many/diff places.  Also, besides the info shows not relevant to these projects itself at the first moment this information probably can be useful for many tooling requirements.

**Comment by estroz** [comment]
The issue and fix in https://github.com/kubernetes-sigs/kubebuilder/pull/1596 are relevant here.

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

I still thinking that it would be simplified if we keep repo == operator name == dir project name by default and in this way, the same layout/standard across all types as suggested before. See the @estroz's suggestion https://github.com/operator-framework/operator-sdk/pull/3341#issuecomment-659066696 as an alternative to have this value in the config instead of getting from any/all places which requires. WDYT? What is your final position? Should we move forward as done here in this PR https://github.com/operator-framework/operator-sdk/pull/3341/commits/75ddd8dfc02fca3179166b56afd134675fe7a1a4 or how can we add this in the config?

File: internal/util/kubebuilder/project.go [hunk 1]
```diff
@@ -50,3 +53,17 @@ func ReadConfig() (*config.Config, error) {
 	}
 	return c, nil
 }
+
```
Comment (camilamacedo86) [inline internal/util/kubebuilder/project.go:56]: Hi @estroz and @joelanford, 

Regardless of the decision that will be made over we keep or not the operator's name in the PROJECT file, this implementation is now centralized here and this would be the only point that could be changed in the future. So, IMO it is not a blocker for we move forward with this PR.

File: internal/util/plugins/makefile.go [hunk 2]
```diff
@@ -97,5 +99,10 @@ bundle: kustomize
 # Build the bundle image.
 bundle-build:
 	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+`
+	// Makefile fragments to add to the base Makefile.
+	makefilePackagemanifests = `
+packagemanifests: kustomize
+	$(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests --version $(VERSION)
 `
```
Comment (camilamacedo86) [inline internal/util/plugins/makefile.go:107]: the target was missing be scaffolded for all types. So it was added as described in the docs. See: https://master.sdk.operatorframework.io/docs/olm-integration/quickstart-package-manifests/#creating-package-manifests

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 3]
```diff
@@ -15,25 +15,72 @@
...
+
+			// fixme: packagemanifests is not running/deploying the controller
+			// see that it just pass in the tests for go because the project/manager is deployed already
+			//By("running the package")
+			//runPkgManCmd := exec.Command(tc.BinaryName, "run", "packagemanifests",
+			//	"--install-mode", "AllNamespaces",
+			//	"--operator-version", operatorVersion,
+			//	"--timeout", "4m")
+			//_, err = tc.Run(runPkgManCmd)
+			//Expect(err).NotTo(HaveOccurred())
```
Comment (camilamacedo86) [inline test/e2e-helm/e2e_helm_olm_test.go:83]: Hi @estroz and @joelanford, 

Now that we split the tests properly was possible to identify that if the project is not deployed on the cluster the  `run packagemanifests` is never succeed.

File: internal/util/plugins/makefile.go [hunk 4]
```diff
@@ -97,5 +99,10 @@ bundle: kustomize
 # Build the bundle image.
 bundle-build:
 	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+`
+	// Makefile fragments to add to the base Makefile.
+	makefilePackagemanifests = `
```
Comment (estroz) [inline internal/util/plugins/makefile.go:104]: 1. We want users to use bundles, not continue using the older package manifests format. Adding this makefile recipe is confusing because now they see two competing options for packaging their operators for OLM.
1. This addition is outside the scope of this PR, so even if (1) weren't the case, I'd still say no to this addition.

File: internal/util/plugins/makefile.go (hunk 2 above)
Comment (estroz) [reply internal/util/plugins/makefile.go:107]: The doc is meant for users who want to _continue_ using package manifests format, not for new OLM users; the docs even say this. That's why there's no `packagemanifests` Makefile target by default.

Comment (camilamacedo86) [reply internal/util/plugins/makefile.go:107]: ok. So we should not scaffold this by default.

File: test/e2e-helm/e2e_helm_olm_test.go (hunk 3 above)
Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go]: I discovered what are the pre-requirements to make it work. Solved.

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 5]
```diff
@@ -15,25 +15,105 @@
...
+
+			By("generating the operator package manifests")
+			err = tc.Make("packagemanifests")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating clusterserviceversion with the manager image")
+			testutils.ReplaceInFile(
+				filepath.Join(tc.Dir, "packagemanifests", operatorVersion,
+					fmt.Sprintf("e2e-%s.clusterserviceversion.yaml", tc.TestSuffix)),
+				"controller:latest", tc.ImageName)
```
Comment (estroz) [inline test/e2e-helm/e2e_helm_olm_test.go:104]: I understand the reasoning behind doing this, but wrapping the same command flow that you'd call in the CLI is pointless. You may as well copy the [Go e2e code](https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite.go#L182-L192) directly.

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:104]: Hi @estroz, 

The implementation suggested will not work for Helm/Ansible and actually IMO we need to change it in GO as well.

Following a few reasons:
a) See that you are calling [tc.KustomizeBuild(filepath.Join("config", "manifests"))](https://github.com/operator-framework/operator-sdk/blob/0c7d59bf34747d305a8ebbd54113ac6330b42239/test/internal/utils.go#L57) it will NOT work for helm/ansible. This projects did not install kustomize with go get since go is not a pre-requirement for them. 

b) See my comment on top of that: 

```
// We will add the target packagemanifests in the Makefile to  be able to run it in tests
// as the users would do it as well which indeed would be harder to executed directly via
// a cmd generated since it requires the kustomize which can be located in the bin of the project
// or in any place in the machine.
```

b) IMO the changes made here are easier to understand as well.

Comment (estroz) [reply test/e2e-helm/e2e_helm_olm_test.go:104]: Helm and Ansible projects will need to install the `kustomize` binary regardless of whether Go is present or not, and this is reflected in their Makefiles. The exact same commands should work for Go, Ansible, Helm:
```sh
$ operator-sdk init
$ operator-sdk create api
$ operator-sdk generate kustomize manifests
$ ${KUSTOMIZE} build config/manifests | operator-sdk generate packagemanifests --version ${VERSION}
```

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:104]: HI @estroz, 

Yes, I agree that the same commands should work for all. However, note that:

- The Go implementation  [Go e2e code](https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite.go#L182-L192) is NOT doing `${KUSTOMIZE} build config/manifests` it is doing `kustomize build config/manifests`
- We should use the makefile targets scaffolded by default instead of calling the commands in order to keep its understand easily and we are able to indeed test them. (same doc steps if possible) It can prevent issues. See that the GO tests are just passing because they are not using the makefile targets Otherwise, it would fail with an error when the `docker-build` would be called. 
- Call `kustomize build config/manifests` as it is done in [Go e2e code](https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite.go#L182-L192) just work for GO because it was installed by `go get dep` which is not possible for Helm/Ansible. 

Sadly. in this specific case I am reluctant to address your suggestion because:
- It will not work for Ansible/Helm since we can not use GO  to install kustomize
- Indeed, IMO we need to change the GO impl because it is harder to understand and keep maintained, not using  the makefile targets, do not work locally when the kustomize is installed in another way)
- IMO we suggest to users customize the Makefile when they need (e.g in this scenario). So, we could do the test following our suggestion which is how it was done. 

I hope that clarifies my motivation here.

Comment (estroz) [reply test/e2e-helm/e2e_helm_olm_test.go:104]: Sounds like we should be installing kustomize into some element of $PATH for Helm and Ansible operators, or locally to `bin/kustomize` for Go operators. Behavior should be consistent.

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:104]: I understand that the behaviour is consistent when we use the Makefile target with `${KUSTOMIZE}` since it will get the kustomize installed. My idea as follow up would cleanup the Go tests as well and ensure that all types are following the same go-tests structure and doing the same common tests.

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: Hi @estroz, 

See that: 
- I updated it to be == we suggest to do it in the docs
- Added the docs links to make clear that
- Add a note in the docs for users know that for ansible and helm the target is == but without the manifests.

Comment (estroz) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: Can we just update $PATH in these tests for now (in `BeforeSuite`) and address your concerns in a follow-up? I agree we should make sure command usage is fairly consistent across all plugins, which requires work outside of the scope of this PR potentially.

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: > Can we just update $PATH in these tests to include ./bin for now (in BeforeSuite)`
No, because the kustomize can be or not in the ./bin and the test should run locally with diff setups. 

Also, see that in order to make consistent we have the GO as well: #3499. Why we would make it work as GO if it is the Go test is not accurate and consistent with the steps described in the docs as it is not working locally?

Comment (joelanford) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: We can run `make kustomize` as a pre-step. At that point `kustomize` is either in `bin/kustomize` OR in the $PATH.

I think @estroz is suggesting that if we add `./bin` to $PATH, then we guarantee that after running `make kustomize`, `kustomize` will be in $PATH (either existing in the user's local setup, or in `./bin`, which has been added to $PATH).

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: Hi @joelanford @estroz, 
 
1 - Why we would call setup the bin PATH for just this test? Again, the kustomzie can or NOT be installed in the bin. The test should work locally.
2 - Why we would call `kustomize build config/manifests` instead to do the test as we suggested for users to do in the docs?

Comment (joelanford) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: Ok so I think we agree that we _can_ do it without appending to the Makefile and running make targets.

The question now is, should we append some targets to the Makefile and run a make target, or should we just make the calls directly.

1. If we use the Makefile approach, we have to add some targets to the Makefile, but then we get the KUSTOMIZE path logic "for free".
1. If we use the Makefile approach, we are testing the [approach that we suggest to users in our docs](https://master.sdk.operatorframework.io/docs/olm-integration/generation/#package-manifests-format).
1. If we use the direct approach, we have to add `<projectDir>/bin` to the PATH.
1. If we use the direct approach, we can call the commands directly, which has less indirection and would make the test code more obvious about what's actually being tested.

Comment (estroz) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: If we're going with the Makefile approach in this PR, I'd like a function to exist that updates the Makefile with some code snippet by appending. The current test code is quite cluttered by this Makefile update.

Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go:116]: IMO appending in the makefile also:
- Makes it easier to read and understand the code and reduces the complexity
- Also, see that by following the suggestion we are not applying the same options suggested in the docs (e.g  see `$(PKG_MAN_OPTS)`)

So, usually, I address all suggestions made but this one I still reluctant because I was not convinced that it will bring advantages and then, I hope that we can reach a consensus that we can keep it as it is. :-)

File: internal/util/projutil/project_util.go [hunk 6]
```diff
@@ -362,5 +364,18 @@ func appendContent(fileContents, target, newContent string) (string, error) {
 
 	index := labelIndex + separationIndex + 1
 	return fileContents[:index] + newContent + fileContents[index:], nil
+}
 
+// GetOperatorName will looking for the value in the config.repo and return the project dir if it do not exist
+// Note that Ansible/Helm operators in the new layout do not have the repo information.
```
Comment (camilamacedo86) [inline internal/util/projutil/project_util.go:370]:
```suggestion
// GetOperatorName will return the name of the operator. Currently, it is the current directory.
```

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 7]
```diff
@@ -15,25 +15,88 @@
...
+				bundleImage += ":" + imageSplit[1]
+			}
+			err = tc.Make("bundle-build", "BUNDLE_IMG="+bundleImage)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("loading the project image into Kind cluster")
+			err = tc.LoadImageToKindClusterWithName(bundleImage)
+			Expect(err).Should(Succeed())
+
+			By("Adding to Makefile target to generate the packagemanifests")
```
Comment (estroz) [inline test/e2e-helm/e2e_helm_olm_test.go:75]:
```suggestion
			By("adding the 'packagemanifests' rule to the Makefile")
```

File: website/content/en/docs/olm-integration/generation.md [hunk 8]
```diff
@@ -134,11 +134,13 @@ endif
...
 
 # Generate package manifests.
-packagemanifests: manifests
+packagemanifests: kustomize manifests
   operator-sdk generate kustomize manifests -q
-  kustomize build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
+  $(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
 ```
 
+**Note** If your project is an Ansible/Helm-based Operator, then the `packagemanifests` works at the same way. However, you need to remove `manifests` target since it do not exist for them. In this way, just replace `packagemanifests: kustomize manifests` with `packagemanifests: kustomize`.
```
Comment (estroz) [inline website/content/en/docs/olm-integration/generation.md:142]: Can you just add a separate rule instead of a note, ex:
````suggestion
```

For Go operator projects:
```make
# Generate package manifests.
packagemanifests: kustomize manifests
  operator-sdk generate kustomize manifests -q
  $(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
```

For other operator project types:
```make
# Generate package manifests.
packagemanifests: kustomize
  operator-sdk generate kustomize manifests -q
  $(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
```
````

**Comment by camilamacedo86** [comment]
Hi @estroz, 

All suggestions applied. Feel free to re-check. Tks for your time in the review.

**Comment by estroz** [comment]
Changes made to `cmd/operator-sdk` code can be removed in this PR, since #3530 changes how operator/project name is retrieved by subcommands.

**Comment by camilamacedo86** [comment]
Hi @estroz kept here the tests and docs since now we have the projectName. 
https://github.com/operator-framework/operator-sdk/pull/3530. Waiting this one get merged :-)

File: test/e2e-helm/e2e_helm_olm_test.go (hunk 5 above)
Comment (camilamacedo86) [reply test/e2e-helm/e2e_helm_olm_test.go]: Closing it since shows solved.

File: internal/util/projutil/project_util.go [hunk 9]
```diff
@@ -351,3 +352,17 @@ func appendContent(fileContents, target, newContent string) (string, error) {
...
 }
+
+// GetOperatorName returns the name of the operator which is by default the projectName attribute of the PROJECT file
+// However, the Go projects built with the plugin version v2 has not this attribute and then, for this case
+// the operatorName will be the current directory.
+func GetOperatorName(cfg *config.Config) (string, error) {
+	if cfg.ProjectName != "" {
+		return cfg.ProjectName, nil
+	}
+	dir, err := os.Getwd()
```
Comment (estroz) [inline internal/util/projutil/project_util.go:363]:
```suggestion
	dir, err := os.Getwd()
	if cfg.IsV3() {
		return errors.New("project config file must contain 'projectName'")
	}
```

File: internal/util/projutil/project_util.go [hunk 10]
```diff
@@ -351,3 +352,17 @@ func appendContent(fileContents, target, newContent string) (string, error) {
...
+	if err != nil {
+		return "", fmt.Errorf("error getting current directory: %v", err)
+	}
+	return strings.ToLower(filepath.Base(dir)), nil
```
Comment (estroz) [inline internal/util/projutil/project_util.go:367]:
```suggestion
	projectName := strings.ToLower(filepath.Base(dir))
	if err := validation.IsDNS1123Label(projectName); err != nil {
		return fmt.Errorf("project name (%s) is invalid: %v", projectName, err)
	}
	return projectName, nil
```

File: internal/plugins/ansible/v1/init.go [hunk 11]
```diff
@@ -112,14 +113,16 @@ func (p *initPlugin) Run() error {
 }
 
 func (p *initPlugin) Validate() error {
-	// Check if the project name is a valid namespace according to k8s
-	dir, err := os.Getwd()
-	if err != nil {
-		return fmt.Errorf("error to get the current path: %v", err)
+	// Check if the project name is a valid k8s namespace (DNS 1123 label).
```
Comment (estroz) [inline internal/plugins/ansible/v1/init.go:116]: These lines are in master already.

File: internal/plugins/ansible/v1/init.go [hunk 12]
```diff
@@ -98,6 +98,7 @@ Optionally creates a new API, using the same flags as "create api"
 func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
 	fs.SortFlags = false
 	fs.StringVar(&p.config.Domain, "domain", "my.domain", "domain for groups")
+	fs.StringVar(&p.config.ProjectName, "project-name", "", "name of this project, the default being directory name")
```
Comment (estroz) [inline internal/plugins/ansible/v1/init.go:101]: This is in master already.

File: internal/util/projutil/project_util.go [hunk 13]
```diff
@@ -351,3 +352,17 @@ func appendContent(fileContents, target, newContent string) (string, error) {
 	return fileContents[:index] + newContent + fileContents[index:], nil
 
 }
+
+// GetOperatorName returns the name of the operator which is by default the projectName attribute of the PROJECT file
+// However, the Go projects built with the plugin version v2 has not this attribute and then, for this case
+// the operatorName will be the current directory.
+func GetOperatorName(cfg *config.Config) (string, error) {
```
Comment (estroz) [inline internal/util/projutil/project_util.go:359]: This function is only ever used in `generate` so move this to `cmd/operator-sdk/generate/internal/genutil.go`.

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 14]
```diff
@@ -15,25 +15,88 @@
...
 		})
 
-		It("Should allow generate the OLM bundle and run it", func() {
+		It("should generate and run a valid OLM bundle and packagemanifests", func() {
+			By("building the bundle")
+			err := tc.Make("bundle")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating the bundle")
+			bundleValidateCmd := exec.Command(tc.BinaryName, "bundle", "validate", "./bundle")
```
Comment (estroz) [inline test/e2e-helm/e2e_helm_olm_test.go:57]: This needs to work universally.
```suggestion
			bundleValidateCmd := exec.Command(tc.BinaryName, "bundle", "validate", "bundle")
```

File: internal/util/projutil/project_util.go (hunk 9 above)
Comment (camilamacedo86) [reply internal/util/projutil/project_util.go]: @estroz the GO projects has not been scaffolded with projectName so it would need to get done in a follow up first. 
Added a todo for that.

Comment (camilamacedo86) [reply internal/util/projutil/project_util.go]: Out of the scope of this PR.

Comment (estroz) [reply internal/util/projutil/project_util.go]: Kubebuilder needs to be [upgraded](https://github.com/operator-framework/operator-sdk/pull/3530/files#diff-37aff102a57d3d7b797f152915a6dc16R46) first.

File: test/e2e-helm/e2e_helm_olm_test.go [hunk 15]
```diff
@@ -15,25 +15,88 @@
...
+			err = tc.Make("install")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("running the package")
+			runPkgManCmd := exec.Command(tc.BinaryName, "run", "packagemanifests",
+				"--install-mode", "AllNamespaces",
+				"--operator-version", operatorVersion,
+				"--timeout", "4m")
+			_, err = tc.Run(runPkgManCmd)
+			Expect(err).NotTo(HaveOccurred())
```
Comment (joelanford) [inline test/e2e-helm/e2e_helm_olm_test.go:99]: Can we separate the `generate/run bundle` tests from the `generate/run packagemanifests` tests?

Can be done in a follow-up

File: website/content/en/docs/olm-integration/generation.md [hunk 16]
```diff
@@ -123,6 +123,29 @@ package manifest YAML file containing channel-to-version mappings, much like a b
...
+ifeq ($(IS_CHANNEL_DEFAULT), 1)
+PKG_IS_DEFAULT_CHANNEL := --default-channel
+endif
+PKG_MAN_OPTS ?= $(FROM_VERSION) $(PKG_CHANNELS) $(PKG_IS_DEFAULT_CHANNEL)
+
+# Generate package manifests.
+packagemanifests: kustomize manifests
+  operator-sdk generate kustomize manifests -q
+  $(KUSTOMIZE) build config/manifests | operator-sdk generate packagemanifests -q --version $(VERSION) $(PKG_MAN_OPTS)
+```
```
Comment (joelanford) [inline website/content/en/docs/olm-integration/generation.md:145]: Ouch - kind of a bummer that we have to duplicate this entire block because of the single `manifests` dependency on the `packagemanifests` target.

This is one of those things that makes organization on our docs site extremely difficult when dealing with three operator types.

I'm almost to the point where I could be convinced to duplicate this (or some portion of this) doc (and other docs with similar issues) for each operator type.

This is okay for now, but definitely something we could think about improving later.

Comment (camilamacedo86) [reply website/content/en/docs/olm-integration/generation.md:145]: I agree that we need to clean up the docs .. keeping it mind for follow-ups.
